sky/skypilot_config.py

```diff
@@ -848,6 +848,7 @@ def is_running_pytest() -> bool:
     global_config_path = _resolve_server_config_path()
     if global_config_path is None:
         global_config_path = get_user_config_path()
+    global_config_path = str(os.path.expanduser(global_config_path))
```

Comment:
this is needed as i think `open()` treats the path as a literal and thus will fail if the path contains a `~` (and it's not expanded).

for example:
```bash
% file ~/.sky/config.yaml
/Users/kevin/.sky/config.yaml: ASCII text
% python3 -c "import os; open(os.path.expanduser('~/.sky/config.yaml'), 'w')"
% python3 -c "open('~/.sky/config.yaml', 'w')"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: '~/.sky/config.yaml'
```